tesseract/tesseract_collision/src/bullet/tesseract_gjk_pair_detector.cpp

```diff
@@ -821,10 +821,10 @@ void TesseractGjkPairDetector::getClosestPointsNonVirtual(const ClosestPointInpu
     }
 
     m_simplexSolver->reset();
-    if (status == 0 && !m_cdata->req.calculate_distance)
+    if (status == 0 && !m_cdata->req.calculate_penetration)
     {
       // The shapes intersect but penetration was not request so return empty contact
-      output.addContactPoint(normalInB, pointOnB + positionOffset, distance);
+      output.addContactPoint(normalInB, pointOnB + positionOffset, -1);
```

Comment:
the variable `distance` defaults to zero but if you set the contact threshold to zero then this was failing to get added as a contact because in the `processResults` function it checks if the distance is less than the contact threshold. This set the distance to negative one, though it may be better to set -1e-30.

Reply:
I'll defer to your judgment here.

Reply:
I will try and see if it works using `-std::numeric_limits<btScalar>::min()`.

Reply:
This worked.